transport/http/client.go

```diff
@@ -47,6 +49,13 @@ func NewClient(
 // ClientOption sets an optional parameter for clients.
 type ClientOption func(*Client)
 
+// ClientResponseFunc takes information from an HTTP request to read from the
+// response and do something additional with the response or add/update
+// context. ClientResponseFuncs are only executed on the client immediately
+// after the request has been made but prior to decoding. This allows for
+// intervention in case there is an error from the request.
+type ClientResponseFunc func(context.Context, *http.Response) context.Context
+
```

Comment:
Please move this definition to request_response_funcs.go, rename the existing ResponseFunc to ServerResponseFunc, and make the comment blocks on {Client,Server}ResponseFunc roughly symmetrical :)